rosidl_generator_c/CMakeLists.txt

```diff
@@ -83,6 +83,11 @@ if(BUILD_TESTING)
     "msg/Wire.msg"
   )
 
+  if(BUILD_SHARED_LIBS)
+    set(${PROJECT_NAME}_LIBRARY_TYPE "SHARED")
+  else()
+    set(${PROJECT_NAME}_LIBRARY_TYPE "STATIC")
+  endif()
```

Comment:
Would moving the block from line 134 before `if(BUILD_TESTING)` fix the issue and avoid duplication of the variable assignment ?

Reply:
Absolutely, I just didn't check "that far down" in the CMake file: 7ba0b8c8960af718a6d2232fa8889d7e2e5c5d0e